oci/layout/oci_dest.go

```diff
@@ -299,6 +303,90 @@ func (d *ociImageDestination) CommitWithOptions(ctx context.Context, options pri
 	return os.WriteFile(d.ref.indexPath(), indexJSON, 0644)
 }
 
+// tryHardlinkLocalFile attempts to hardlink the specified file and digest it.
+func tryHardlinkLocalFile(dest *ociImageDestination, file string) (private.UploadedBlob, bool, error) {
+	blobFile, err := os.CreateTemp(dest.ref.dir, "oci-put-blob")
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+
+	// Remove the file for os.Link() to work.
+	blobFile.Close()
+	os.Remove(blobFile.Name())
+	if err := os.Link(file, blobFile.Name()); err != nil {
```

Comment:
I’m quite sure about a hard-link to the user’s input; the user could continue to edit that file afterwards. At the very least this would need a strong API documentation.

I think a reflink would be uncontroversial, OTOH not always possible.

Reply:
Fair point.  Reflinks work on Linux (xfs, btrfs) and on Mac OS (? see [unix package](https://pkg.go.dev/golang.org/x/sys/unix#IoctlFileClone)).  Hardlinks work everywhere, so they are more portable.

If we envision `podman artifact` having to deal with huge files (e.g., LLMs), then hardlinks seem more attractive.  On the other hand, any change to the file after a `podman artifact add` would corrupt the image as the digest wouldn't match anymore.  The digest could be recomputed on `artifact push` for instance.

OR will `podman artifact` on the remote-clients be talking to podman-machine?

@baude can you give direction?

Reply:
Had a chat with @baude.  Reflinks it will be as the artifact store must live on a Linux machine for volume-mounts etc. to work.  Will massage the PR tomorrow morning.